src/main/java/core/basesyntax/service/strategy/OperationProcessorImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.model.Operation;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OperationProcessorImpl implements OperationProcessor {
+    private final Map<Operation, OperationHandler> operationHandlers = new HashMap<>();
+
+    public void addOperationHandler(Operation operation, OperationHandler handler) {
+        operationHandlers.put(operation, handler);
+    }
+
+    public void processOperation(Operation operation) {
+        OperationHandler handler = operationHandlers.get(operation);
+        if (handler != null) {
+            handler.calculateBalanceAfterActivities();
+        }
+    }
+}

```

Comment:
1. Your OperationProcessore is basically core for strategy pattern. It's purpose is to provide proper implementation
2. Please accept ready to use map in constructor
```suggestion

public class OperationProcessorImpl implements OperationProcessor {
    private final Map<Operation, OperationHandler> handlersByOperation;

    public OperationProcessorImpl(Map<Operation, OperationHandler>handlersByOperation) {
        this.handlersByOperation = handlersByOperation;
    }

    public OperationHandler getHandler(Operation operation) {
        OperationHandler handler = operationHandlers.get(operation);
    }
}
```